Modules/Loadable/SubjectHierarchy/Widgets/qSlicerSubjectHierarchyPluginHandler.cxx

```diff
@@ -117,6 +117,17 @@ qSlicerSubjectHierarchyPluginHandler::~qSlicerSubjectHierarchyPluginHandler()
   delete this->m_DefaultPlugin;
 }
 
+//---------------------------------------------------------------------------
+bool qSlicerSubjectHierarchyPluginHandler::unRegisterPlugin(QString name)
+{
+  qSlicerSubjectHierarchyAbstractPlugin* currentPlugin = pluginByName(name);
+  foreach(QAction* action, currentPlugin->viewContextMenuActions())
+    {
+    this->m_PluginLogic->unRegisterViewMenuAction(action);
```

Comment:
This function does not actually unregister plugins, it only removes the view actions provided by the given plugin! There are much more to plugins than doing this  (there are other types of actions, there is a list of registered plugins in this class etc). Please look at what `registerPlugin` does (and the abstract SH plugin class to see what they provide and what needs to be removed) and finish this function. Or rename it to make it obvious what it does and what it doesn't.